src/app/features/tasks/tasks-related-to-entity/tasks-related-to-entity.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component } from "@angular/core";
+import { OnInitDynamicComponent } from "../../../core/view/dynamic-components/on-init-dynamic-component.interface";
+import { FormFieldConfig } from "../../../core/entity-components/entity-form/entity-form/FormConfig";
+import { Entity } from "../../../core/entity/model/entity";
+import { PanelConfig } from "../../../core/entity-components/entity-details/EntityDetailsConfig";
+import { Todo } from "../model/todo";
+import { DatabaseIndexingService } from "../../../core/entity/database-indexing/database-indexing.service";
+import { HistoricalEntityData } from "../../historical-data/model/historical-entity-data";
+import { DynamicComponent } from "../../../core/view/dynamic-components/dynamic-component.decorator";
+
+@DynamicComponent("TasksRelatedToEntity")
+@Component({
+  selector: "app-tasks-related-to-entity",
+  templateUrl: "./tasks-related-to-entity.component.html",
+  styleUrls: ["./tasks-related-to-entity.component.scss"],
+})
+export class TasksRelatedToEntityComponent implements OnInitDynamicComponent {
+  entries: Todo[] = [];
+
+  columns: FormFieldConfig[] = [
+    { id: "deadline" },
+    { id: "subject" },
+    { id: "assignedTo" },
+    { id: "description", visibleFrom: "xl" },
+    { id: "repetitionInterval", visibleFrom: "xl" },
+    { id: "relatedEntities", hideFromTable: true },
+  ];
+
+  private entity: Entity;
+
+  /** the property name of the Todo that contains the ids referencing related entities */
+  private referenceProperty: string = "relatedEntities";
+
+  constructor(private dbIndexingService: DatabaseIndexingService) {
+    this.createIndex();
+  }
+
+  // TODO: filter by current user as default in UX? --> custom filter component or some kind of variable interpolation?
+
+  async onInitFromDynamicConfig(config: PanelConfig) {
+    this.entity = config.entity;
+    this.columns = config.config?.columns ?? this.columns;
+
+    this.entries = await this.loadDataFor(this.entity.getId(true));
+  }
+
+  private createIndex() {
+    // TODO: move this generic index creation into schema + indexing service
+
+    const designDoc = {
+      _id: "_design/todo_index",
+      views: {
+        by_entity: {
+          map: `(doc) => {
+            if (!doc._id.startsWith("${Todo.ENTITY_TYPE}")) return;
+            if (!Array.isArray(doc.${this.referenceProperty})) return;
+
+            var d = new Date(doc.deadline || null);
+            var dateString = d.getFullYear() + "-" + String(d.getMonth()+1).padStart(2, "0") + "-" + String(d.getDate()).padStart(2, "0")
```

Comment:
Let's use `date-only` and then the date string already has a good format (e.g. `2022-01-29`)

Reply:
Or do you already have a timebound todo in mind?

Reply:
oh wow, I never thought about the fact that those dates are already stored as string! This is actually copy&paste from other indices but you are right and this should actually speed up the index calculations, I think.

While reminders at a certain time of day might be interesting, I think I would leave this aside in the deadline for now ...

Reply:
What about records with an undefined date here? Currently these default to 1970-01-01  shall we set that as a string then?